### ISSUE TITLE
wrong placement of semicolons in the output of the dereplication command

### ISSUE THREAD

**Issue by frederic-mahe** [torognes/vsearch#323]
@torognes Something is wrong with vsearch's dereplication. The semicolons are misplaced in the output, creating bugs in downstream pipelines!

With abundance annotations in the input, without `--sizein`:
```sh
printf ">s1;size=2;\nA\n>s2;size=1;\nA\n" | \
    vsearch --derep_fulllength - --minseqlength 1 --quiet --sizeout --output -
```
```
>s1;;size=2
A
```
With abundance annotations in the input, with `--sizein`:
```sh
printf ">s1;size=2;\nA\n>s2;size=1;\nA\n" | \
    vsearch --derep_fulllength - --minseqlength 1 --quiet --sizein --sizeout --output -
```
```
>s1;;size=3
A
```
The correct output should be:
```
>s1;size=3;
A
```

To cover all cases, here are the results without abundance annotation in the input, without `--sizeout`: 
```sh
printf ">s1\nA\n>s2\nA\n" | \
    vsearch --derep_fulllength - --minseqlength 1 --quiet --output -
```
```
>s1
A
```
Without abundance annotation in the input, with `--sizeout`: 
```sh
printf ">s1\nA\n>s2\nA\n" | \
    vsearch --derep_fulllength - --minseqlength 1 --quiet --sizeout --output -
```
```
>s1;size=2
A
```
With abundance annotations in the input, with `--sizein` but no `--sizeout`:
```sh
printf ">s1;size=2;\nA\n>s2;size=1;\nA\n" | \
    vsearch --derep_fulllength - --minseqlength 1 --quiet --sizein --output -
```
```
>s1;size=2;
A
```

**Comment by torognes** [comment]
This error has been fixed in version 2.8.2. If the original header was correct, the updated header should be correct too. Note that vsearch will use semicolons to separate the parts of the header and will not place one at the end of the header (e.g. `>abc;size=1` not `>abc;size=1;`).

**Comment by frederic-mahe** [comment]
Awesome, I have added five new tests to `vsearch-test` to cover that specific issue.
